src/main/java/seedu/address/logic/commands/AutoGradeCommand.java

```diff
@@ -0,0 +1,129 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_AUTOGRADE_TYPE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PASSING_GRADE;
+
+import java.util.Arrays;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.student.Student;
+import seedu.address.model.student.StudentGrade;
+import seedu.address.model.student.model.StudentBook;
+
+
+/**
+ * Format full help instructions for every command for display.
+ */
+public class AutoGradeCommand extends Command {
+    public static final String COMMAND_WORD = "autoGrade";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Automatically grade students based on total score.\n"
+            + "Parameters:\n"
+            + PREFIX_AUTOGRADE_TYPE + "GRADING_METHOD "
+            + PREFIX_PASSING_GRADE + "PASSING_GRADE_VALUE \n"
+            + "Grading method: Must be one of \"p\", \"percentile\", \"Percentile\", \"a\", "
+            + "\"absolute\", \"Absolute\"\n"
+            + "Passing grade value: must be float and must be filled at most with 11 values.\n"
+            + "Values are in format of " + PREFIX_PASSING_GRADE + "[A+] [A] [A-] [B+] [B] [B-] [C+] [C] [D+] [D] [F] \n"
+            + "Example: " + COMMAND_WORD + " " + PREFIX_AUTOGRADE_TYPE + "percentile "
+            + PREFIX_PASSING_GRADE + "90 80 70 60 50 40 30 20 18 15 12";
```

Comment:
If the grade list is the same length as the percentile/absolute list, then what to assign students who are below the last value? Prob shoud include a default "U" or smthing